scrollingpagerindicator/src/main/java/ru/tinkoff/scrollingpagerindicator/RecyclerViewAttacher.java

```diff
@@ -38,30 +38,53 @@
      * +------------------------------+
      */
     public RecyclerViewAttacher() {
-        currentPageLeftCornerX = 0; // Unused when centered
-        centered = true;
+        this(null);
     }
 
     /**
      * Use this constructor if current page in recycler isn't centered.
      * All pages must have the same width.
      * Like this:
-     *
+     * <p>
      * +-|----------------------------+
      * | +--------+  +--------+  +----|
      * | | current|  |        |  |    |
      * | |  page  |  |        |  |    |
      * | +--------+  +--------+  +----|
      * +-|----------------------------+
-     *   | currentPageLeftCornerX
-     *   |
+     * | currentPageLeftCornerX
+     * |
+     *
      * @param currentPageLeftCornerX x coordinate of current view left corner relative to recycler view.
      */
     public RecyclerViewAttacher(int currentPageLeftCornerX) {
+        this(currentPageLeftCornerX, null);
+    }
+
+    /**
+     * @param currentPageLeftCornerX x coordinate of current view left corner relative to recycler view.
+     * @param adapterDelegate        is a delegate which is an abstraction under RecyclerView's Adapter
+     *                               which helps you to use interact with Adapter in some tricky way
+     * @see {@link #RecyclerViewAttacher(int currentPageLeftCornerX)}
+     */
+    public RecyclerViewAttacher(int currentPageLeftCornerX, RecyclerViewAdapterDelegate adapterDelegate) {
         this.currentPageLeftCornerX = currentPageLeftCornerX;
         this.centered = false;
+        this.adapterDelegate = adapterDelegate;
     }
 
+    /**
+     * @param adapterDelegate is a delegate which is an abstraction under RecyclerView's Adapter
+     *                        which helps you to use interact with Adapter in some tricky way
+     * @see {@link #RecyclerViewAttacher()} default constructor for more info
+     */
+    public RecyclerViewAttacher(RecyclerViewAdapterDelegate adapterDelegate) {
+        currentPageLeftCornerX = 0; // Unused when centered
+        centered = true;
```

Comment:
Add `this.` for these fields too for common style

Reply:
done